### PR TITLE
Update test_utils with ThreadPool test

### PR DESCRIPTION
**Pull request by cydrain**
Issue: #809

### CODE REVIEW

File: tests/ut/test_utils.cc [hunk 1]
```diff
@@ -133,6 +133,50 @@ TEST_CASE("Test DiskLoad") {
...
+
+        REQUIRE(knowhere::ThreadPool::GetSearchThreadPoolPendingTaskCount() == 0);
+
+        if (prev_search_thread_num > 0) {
+            knowhere::ThreadPool::SetGlobalSearchThreadPoolSize(prev_search_thread_num);
+        }
+    }
+
+    SECTION("ScopedOmpSetter") {
+        knowhere::ThreadPool::ScopedOmpSetter setter(1);
```
[the remark this answers is not in the document]
Comment (cydrain) [reply tests/ut/test_utils.cc:176]: To let coverage can cover these lines
Screenshot from 2024-09-09 10-18-18

[the remark this answers is not in the document]
Comment (zhengbuqian) [reply tests/ut/test_utils.cc:176]:
```
SECTION("ScopedOmpSetter") {
        int initial = omp_get_max_threads();
        {
            knowhere::ThreadPool::ScopedOmpSetter setter(2 * initial);
            assert omp_get_max_threads() == 2 * initial;
        }
        assert omp_get_max_threads() == initial;

}
```

[the remark this answers is not in the document]
Comment (cydrain) [reply tests/ut/test_utils.cc]: updated

File: tests/ut/test_utils.cc [hunk 2]
```diff
@@ -133,6 +133,57 @@ TEST_CASE("Test DiskLoad") {
...
+
+        REQUIRE(knowhere::ThreadPool::GetSearchThreadPoolPendingTaskCount() == 0);
+
+        if (prev_search_thread_num > 0) {
+            knowhere::ThreadPool::SetGlobalSearchThreadPoolSize(prev_search_thread_num);
+        }
+    }
+
+    SECTION("ScopedOmpSetter") {
+        int prev_num_threads = omp_get_num_threads();
```
[the remark this answers is not in the document]
Comment (cydrain) [reply tests/ut/test_utils.cc:176]: correct

**Pull request merged**